### PR TITLE
add pse

### PR DESCRIPTION
**Pull request by WenmuZhou**
实现PSE算法
[论文](https://arxiv.org/abs/1903.12473v2),[官方repo](https://github.com/whai362/PSENet)

指标情况
所有测试均在V100下完成，输入图像缩放到短边736
FPS只计算了前向和后处理

|模型|precision|recall|Hmean|FPS|
| --- | --- | --- | --- | --- | 
|resnet50(官方)|83.6%|	74.0%|78.5%|14.3|
|ResNet50_vd|85.81%|79.53%|82.55%|16.51|
|MobileNetV3|82.20%|70.48%|75.89%|21.19|

### CODE REVIEW

File: ppocr/losses/det_pse_loss.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+# @Time    : 3/29/19 11:03 AM
+# @Author  : zhoujun
```
Comment (littletomatodonkey) [inline ppocr/losses/det_pse_loss.py:3]: 换成Paddlepaddle的license

File: ppocr/losses/det_pse_loss.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+        input = input.reshape([input.shape[0], -1])
+        target = target.reshape([target.shape[0], -1])
+        mask = mask.reshape([mask.shape[0], -1])
+
+        input = input * mask
+        target = target * mask
+
+        a = paddle.sum(input * target, 1)
+        b = paddle.sum(input * input, 1) + 0.001
```
Comment (littletomatodonkey) [inline ppocr/losses/det_pse_loss.py:79]: 0.001是什么？eps？这块建议写成带默认值的可传入参数吧

File: ppocr/losses/det_pse_loss.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        c = paddle.sum(target * target, 1) + 0.001
+        d = (2 * a) / (b + c)
+        return 1 - d
+
+    def ohem_single(self, score, gt_text, training_mask, ohem_ratio=3):
+        pos_num = int(paddle.sum((gt_text > 0.5).astype('float32'))) - int(
+            paddle.sum(paddle.logical_and((gt_text > 0.5), (training_mask <= 0.5)).astype('float32')))
+
+        if pos_num == 0:
+            # selected_mask = gt_text.copy() * 0 # may be not good
```
Comment (littletomatodonkey) [inline ppocr/losses/det_pse_loss.py:89]: 删掉

File: ppocr/metrics/eval_det_iou.py [hunk 4]
```diff
@@ -168,22 +168,27 @@ def compute_ap(confList, matchList, numGtCare):
         numGlobalCareGt += numGtCare
         numGlobalCareDet += numDetCare
 
+        # perSampleMetrics = {
```
Comment (littletomatodonkey) [inline ppocr/metrics/eval_det_iou.py:171]: 删掉

File: ppocr/losses/det_pse_loss.py (hunk 1 above)
Comment (WenmuZhou) [reply ppocr/losses/det_pse_loss.py]: done tks

File: ppocr/losses/det_pse_loss.py (hunk 2 above)
Comment (WenmuZhou) [reply ppocr/losses/det_pse_loss.py]: done tks

File: ppocr/losses/det_pse_loss.py (hunk 3 above)
Comment (WenmuZhou) [reply ppocr/losses/det_pse_loss.py]: done ,tks

File: ppocr/metrics/eval_det_iou.py (hunk 4 above)
Comment (WenmuZhou) [reply ppocr/metrics/eval_det_iou.py]: done ,tks

**Review by MissPenguin — APPROVED** [review]

**Pull request merged**